### PR TITLE
Sync what the bot thinks are required status with github

### PR DESCRIPTION
**Pull request by eparis**
This will sync required contexts, retest required contexts, and extra
contexts (possibly used for things like the cla context) between the bot
and github automatically.

### CODE REVIEW

**Comment by lavalamp** [comment]
I am on board with the concept. @foxish do you have time to review?

**Comment by foxish** [comment]
@lavalamp, yes, I can review this later today.

**Comment by eparis** [comment]
I'll fix CI in a minute. stoopid Eric. stoopid.

File: mungegithub/github/github.go [hunk 1]
```diff
@@ -533,6 +540,85 @@ func (config *Config) deleteCache(resp *github.Response) {
...
+		},
+	}
+	_, _, err := config.client.Repositories.EditBranch(config.Org, config.Project, name, branch)
+	if err != nil {
+		glog.Errorf("Unable to edit branch protections for %s: %v", name, err)
+	}
+	return err
+}
+
+// needsBranchProtection returns if the branch is protected by exactly the set of
```
Comment (foxish) [inline mungegithub/github/github.go]: "returns true"?

File: mungegithub/submit-queue/deployment/kubernetes/configmap.yaml [hunk 2]
```diff
@@ -14,8 +14,11 @@ data:
   # test-options feature options.
   test-options.required-retest-contexts: "\"Jenkins GCE e2e\",\"Jenkins unit/integration\",\"Jenkins verification\",\"Jenkins GCE Node e2e\",\"Jenkins Kubemark GCE e2e\",\"Jenkins GCE etcd3 e2e\""
 
+  submit-queue.protected-branches: "master"
+  submit-queue.protected-branches-extra-contexts: "cla/linuxfoundation"
```
Comment (eparis) [inline mungegithub/submit-queue/deployment/kubernetes/configmap.yaml]: The bot can override these, well assuming it has admin privs in the repo....

Comment (foxish) [inline mungegithub/submit-queue/deployment/kubernetes/configmap.yaml]: Currently it has only "write" privileges. @thockin, ok to grant k8s-merge-robot admin in the kubernetes-org (or the repos with the merge-bot active?)

Comment (thockin) [inline mungegithub/submit-queue/deployment/kubernetes/configmap.yaml]: Does it have 2FA activated?  I don't know what this PR does or what privs it needs..

Comment (foxish) [inline mungegithub/submit-queue/deployment/kubernetes/configmap.yaml]: Filed https://github.com/kubernetes/contrib/issues/2129. @eparis, for now, maybe we could exclude the CLA status from being marked as required perhaps?

**Pull request merged**